### PR TITLE
add a function to get default interface

### PR DESCRIPTION
**Pull request by smruti77**
add get_default_interface() in interface.py to get default
interface name

Signed-off-by: bismurti bidhibrata pattajoshi <bbidhibr@in.ibm.com>

### CODE REVIEW

**Review by willianrampazzo — COMMENTED** [review]
I noticed the `ip` command also has the `-j` option for JSON output. Doesn't that make things easy? You will not need to handle splits of multiples lines or splits of a line to get the values you need. Just a suggestion here.

File: avocado/utils/network/interfaces.py [hunk 1]
```diff
@@ -176,6 +176,19 @@ def is_link_up(self):
         """
         return self.is_admin_link_up() and self.is_operational_link_up()
 
+    def get_default_interface(self):
```
Comment (beraldoleal) [inline avocado/utils/network/interfaces.py:179]: to avoid ambiguity, I suggest calling this  `get_default_route_iface` or `get_default_route_interface()`.

**Review by beraldoleal — CHANGES_REQUESTED** [review]
Hi @smruti77 this is welcome, but I left a few comments for you.

File: avocado/utils/network/interfaces.py [hunk 2]
```diff
@@ -176,6 +176,19 @@ def is_link_up(self):
...
+        """Get the default interface.
+
+        :return: Default interface name.
+        """
+        output = run_command("ip route", self.host)
```
Comment (PaulYuuu) [inline avocado/utils/network/interfaces.py:184]: You can use `ip route list default`

Comment (willianrampazzo) [reply avocado/utils/network/interfaces.py:184]: Indeed, `ip -j route list default` is a good option here.

[the remark this answers is not in the document]
Comment (smruti77) [reply avocado/utils/network/interfaces.py:184]: ok

File: avocado/utils/network/interfaces.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (smruti77) [reply avocado/utils/network/interfaces.py:179]: in Host() will be better.

File: avocado/utils/network/interfaces.py (hunk 2 above)
Comment (smruti77) [reply avocado/utils/network/interfaces.py:184]: ok, will do that.

**Comment by smruti77** [comment]
#4245

**Pull request closed without merging**